File: Arrays and Hashing/14-Longest-Common-Prefix/14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        new = ""
        strs = sorted(strs)
        first, last = strs[0], strs[-1]
        for i in range(len(first)):
            if i < len(last) and first[i] == last[i]:
                new += first[i]
            else:
                break
        return new

File: Arrays and Hashing/14-Longest-Common-Prefix/14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(!strs.size()){
            return "";
        }
        string prefix = "";
        sort(strs.begin(), strs.end());
        string first = strs[0];
        string last = strs.back();
        for(int i = 0; i < first.size(); i++){
            if(i < last.size() && first[i] == last[i]){
                prefix += first[i]; 
            }
            else{
                break;
            }
        }
        return prefix;
    }
};